ml-prims/src/stats/minmax.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "cuda_utils.h"
+#include <limits>
+
+
+namespace MLCommon {
+namespace Stats {
+
+///@todo: implement a proper "fill" kernel
+template <typename T>
+__global__ void minmaxInitKernel(int ncols, T* globalmin, T* globalmax,
+                                 T init_val) {
+    int tid = threadIdx.x + blockIdx.x * blockDim.x;
+    if(tid >= ncols)
+        return;
+    globalmin[tid] = init_val;
+    globalmax[tid] = -init_val;
+}
+
+template <typename T>
+__global__ void minmaxKernel(const T* data, const int* rowids,
+                             const int* colids, int nrows, int ncols,
+                             T* g_min, T* g_max, T* sampledcols,
+                             T init_min_val) {
+    int tid = threadIdx.x + blockIdx.x * blockDim.x;
+    extern __shared__ char shmem[];
+    T *s_min = (T*)shmem;
+    T *s_max = (T*)(shmem + sizeof(T) * ncols);
+    for (int i = threadIdx.x; i < ncols; i += blockDim.x) {
+        s_min[i] = init_min_val;
+        s_max[i] = -init_min_val;
+    }
+    __syncthreads();
+    for (int i = tid; i < nrows*ncols; i += blockDim.x*gridDim.x) {
+        int col = i / nrows;
+        int row = i % nrows;
+        if(colids != nullptr) {
+            col = colids[col];
+        }
+        if(rowids != nullptr) {
+            row = rowids[row];
+        }
+        if(row < 0 || col < 0) {
+            continue;
+        }
+        int index = row + col * nrows;
```

Comment:
For the use case in RF, we would have shuffled columns, so we do index = row + colids[col] * nrows, where colids is an array of shuffled columns numbers [0,Ncols.)

Reply:
@vishalmehta1991 for your point number 1, lines 55 and 58 would be sufficient, right? or do you think I'm still missing some logic here?

for your point number 2, I have dropped a question in the original PR #460

Reply:
For point 2, i think we would need nrowsoffset, to allow jumps in original data resident in the GPU. 
we needs this because nrows and ncols passed int the kernel, are size of sampled data.

Reply:
Our discussion in PR #460 has cleared all my doubts about the usage of these 'indexing' arrays. Let me update this prim accordingly.

Reply:
@vishalmehta1991, I have updated the prim as per our discussion here. Can you check now?

Reply:
Looks good now, we can use this ml-prim in RF now.